lib/frame.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require_relative './roll'
+
+class Frame
+  attr_reader :pins_left, :result, :score
+
+  def initialize
```

Comment:
Clearly defined and separate Frame class. I can see clearly what's initialised and stored here.